src/main/java/htsjdk/variant/variantcontext/JEXLMap.java

```diff
@@ -131,19 +136,17 @@ private void initialize(Collection<JexlVCMatchExp> jexlCollection) {
      *                                  when the Jexl expression in {@code exp} fails to evaluate the JexlContext
      *                                  constructed with the input VC or genotype.
      */
-    private void evaluateExpression(final JexlVCMatchExp exp) {
+    private boolean evaluateExpression(final JexlVCMatchExp exp) {
         // if the context is null, we need to create it to evaluate the JEXL expression
         if (this.jContext == null) {
-            createContext();
+            jContext = createContext();
         }
 
         try {
             final Boolean value = (Boolean) exp.exp.evaluate(jContext);
-            // treat errors as no match
-            jexl.put(exp, value == null ? false : value);
+            return value == null ? howToTreatMissingValues.getMissingValue() : value;
         } catch (final JexlException.Variable e) {
-            // if exception happens because variable is undefined (i.e. field in expression is not present), evaluate to FALSE
-            jexl.put(exp,false);
+            return howToTreatMissingValues.getMissingValue();
```

Comment:
This critical try/catch definitely needs a couple lines of explanatory comments for posterity. In particular you need to document what would cause a `JexlException.Variable` exception to be thrown, and you need to document the fact that `howToTreatMissingValues.getMissingValue()` can itself throw.